octodns/manager.py

```diff
@@ -244,6 +227,37 @@ def __init__(self, config_file, max_workers=None, include_meta=False):
                     'Incorrect plan_output config for ' + plan_output_name
                 )
 
+        self._zone_tree = None
+
+    @property
+    def zone_tree(self):
+        if self._zone_tree is None:
+            zone_tree = {}
+
+            # Get a list of all of our zone names
+            zones = list(self.config['zones'].keys())
+            # Sort them from shortest to longest so that parents will always
+            # come before their subzones
+            zones.sort(key=lambda z: len(z))
+            # Until we're done processing zones
+            while zones:
+                # Grab the one we'lre going to work on now
+                zone = zones.pop(0)
+                trimmer = len(zone) + 1
+                subs = set()
+                # look at all the zone names that come after it
+                for candidate in zones:
+                    # If they end with this zone's name them they're a sub
+                    if candidate.endswith(zone):
```

Comment:
I think this is part of the original bug, if I pass `ssub.unit.tests.` and `sssub.unit.tests.` through as zones I get the following:
```
{
    ...,
    'sub.unit.tests.': {'', 'another', 's'},
    'ssub.unit.tests.': {''},
    'sssub.unit.tests.': set(),
    ...,
```
That's definitely not right, as the tree would see `unit.tests.` have three children, `sub, ssub, sssub`.

Reply:
Yeah. That's pretty much the same issue you found in https://github.com/octodns/octodns/pull/918

Reply:
Added a test to hit this problem. Looking at solutions now.

Reply:
This one was trivial, just needed to search for endswith the zone name with a dot prepended. It now work similar to `add_record`'s check so if one is understood the other will make more sense.